### ISSUE TITLE
Lost ability to search across multiple fields after upgrade to 1.1.2

### ISSUE THREAD

**Issue by coryschires** [kidpollo/tanker#34]
Let's say I have a user:

```
+------------+-----------+------------+-----------+
| first_name | last_name | department | position  |
+------------+-----------+------------+-----------+
| Joe        | Smith     | History    | Professor |
+------------+-----------+------------+-----------+
```

When running version 1.0.0...

```
User.search_tank("Joe")
User.search_tank("Smith")
User.search_tank("Joe Smith")
```

...would all return the model instance for Joe Smith. After upgrading to version 1.1.2, I can still find Joe with...

```
User.search_tank("Joe")
User.search_tank("Smith")
```

but... 

```
User.search_tank("Joe Smith")
```

...returns an empty array. Any idea what might be going on?

PS: New snippets are working great for me. Thanks a bunch! You should add a donate button to the repo.

**Comment by kidpollo** [comment]
I actually did some changes on how the search query is produced in order to introduce the snippeting. I have been corresponding with the index tank ceo and working directly with their devs. I will look into this in the following days. Sorry its been kinda crazy lately. I have a pretty good idea on how to fix this.

I will also add more integration tests to cover this issue.

**Comment by coryschires** [comment]
Awesome. I actually have another issue related to snippets. It seems that when returning a collection, the snippets for the last item in the collection appear to overwrite the previous item's snippets. I'll look more into this and probably write some integration tests to replicate the issue.

I also had a look at the integration tests and the coverage looks pretty slim around snippets. I'd like to use this gem for an app that's due to release within a few weeks. So I'm definitely interested in helping work thru any remaining bugs. I plan to give the code a closer read this afternoon. 

If at all possible, keep me in the loop as you continue to discuss plans with the index tank people so that I have a good idea where best to direct my efforts.

Thanks again!

**Comment by kidpollo** [comment]
Yes for sure!

First stop is making tanker more stable and have more tests that make it less brittle across big changes. 

A quick roadmap for now would be (in that order):

Conditional indexing (per instance and per attribute)
Categories and faceting
Category filtering
Better dsl for Variables and functions
Clean dsl for GEO
JS powered realtime search helpers :D

Thats it for now but Index Tank is awesome to work with and I hope to add mor things to help on leveraging Index Tank as a data visualization analytics source not just a search service.

**Comment by coryschires** [comment]
I have confirmed that snippets are in fact being overwritten with a failing integration test.

Incidentally, I'm in the process of overhauling the integration tests in order to better catch some of these issues during future releases. Along the way, I've uncovered a few serious problems with how the tests were written which explains why these bugs slipped thru the cracks. I'll give more details in the pull request once I'm finished poking around.

As always, thanks for the awesome gem.
